tiledb/sm/serialization/array_schema.cc

```diff
@@ -206,7 +206,7 @@ tuple<Status, optional<shared_ptr<Filter>>> filter_from_capnp(
               tiledb::common::make_shared<FloatScalingFilter>(HERE())};
     }
     case FilterType::FILTER_NONE:
-      break;
+      return {Status::Ok(), std::nullopt};
```

Comment:
This should return an instance of `NoopFilter`. `FILTER_NONE` refers to this filter, not to no filter at all.